File: win-prog/ProgWinVB/String Theory/StringAppend/StringAppend.vb
'---------------------------------------------
' StringAppend.vb (c) 2002 by Charles Petzold
'---------------------------------------------
Imports System

Module StringAppend
    Const iIterations As Integer = 10000

    Sub Main()
        Dim dt As DateTime = DateTime.Now
        Dim str As String = String.Empty
        Dim i As Integer

        For i = 1 To iIterations
            str &= "abcdefghijklmnopqrstuvwxyz" & vbCrLf
        Next i

        Console.WriteLine(DateTime.op_Subtraction(DateTime.Now, dt))
    End Sub
End Module

File: win-prog/ProgWinCS/String Theory/StringAppend/StringAppend.cs
//-------------------------------------------
// StringAppend.cs © 2001 by Charles Petzold
//-------------------------------------------
using System;

class StringAppend
{
     const int iIterations = 10000;

     public static void Main()
     {
          DateTime dt  = DateTime.Now;
          string   str = String.Empty;

          for (int i = 0; i < iIterations; i++)
               str += "abcdefghijklmnopqrstuvwxyz\r\n";

          Console.WriteLine(DateTime.Now - dt);
     }
}